### PR TITLE
Sovereign shard coordinator

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Created sovereign shard coordinator
  
## Proposed changes
- Created sovereign factories and integrated them

## Testing procedure
- Tested with this branch: `sovereign-test-ugly-shard`

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by sasurobert — DISMISSED** [review]

File: genesis/process/genesisBlockCreator.go [hunk 1]
```diff
@@ -16,13 +16,13 @@ import (
 	"github.com/multiversx/mx-chain-go/config"
 	"github.com/multiversx/mx-chain-go/dataRetriever"
 	"github.com/multiversx/mx-chain-go/dataRetriever/blockchain"
+	errorsMx "github.com/multiversx/mx-chain-go/errors"
```
Comment (SebastianMarian) [inline genesis/process/genesisBlockCreator.go:19]: Alias could be removed

File: update/factory/exportHandlerFactory.go [hunk 2]
```diff
@@ -15,6 +15,7 @@ import (
 	"github.com/multiversx/mx-chain-go/epochStart"
 	"github.com/multiversx/mx-chain-go/epochStart/notifier"
 	"github.com/multiversx/mx-chain-go/epochStart/shardchain"
+	errorsMx "github.com/multiversx/mx-chain-go/errors"
```
Comment (SebastianMarian) [inline update/factory/exportHandlerFactory.go:18]: Alias could be removed

File: update/factory/fullSyncInterceptors.go [hunk 3]
```diff
@@ -7,6 +7,7 @@ import (
 	"github.com/multiversx/mx-chain-core-go/marshal"
 	"github.com/multiversx/mx-chain-go/common"
 	"github.com/multiversx/mx-chain-go/dataRetriever"
+	errorsMx "github.com/multiversx/mx-chain-go/errors"
```
Comment (SebastianMarian) [inline update/factory/fullSyncInterceptors.go:10]: Alias could be removed

File: update/factory/fullSyncRequestersContainerFactory.go [hunk 4]
```diff
@@ -11,6 +11,7 @@ import (
 	"github.com/multiversx/mx-chain-go/dataRetriever/requestHandlers/requesters"
 	"github.com/multiversx/mx-chain-go/dataRetriever/topicSender"
 	"github.com/multiversx/mx-chain-go/epochStart/bootstrap/disabled"
+	errorsMx "github.com/multiversx/mx-chain-go/errors"
```
Comment (SebastianMarian) [inline update/factory/fullSyncRequestersContainerFactory.go:14]: Alias could be removed

File: genesis/process/genesisBlockCreator.go (hunk 1 above)
Comment (mariusmihaic) [reply genesis/process/genesisBlockCreator.go:19]: Right, removed

File: update/factory/exportHandlerFactory.go (hunk 2 above)
Comment (mariusmihaic) [reply update/factory/exportHandlerFactory.go:18]: Right, removed

File: update/factory/fullSyncInterceptors.go (hunk 3 above)
Comment (mariusmihaic) [reply update/factory/fullSyncInterceptors.go]: Right, removed

File: update/factory/fullSyncRequestersContainerFactory.go (hunk 4 above)
Comment (mariusmihaic) [reply update/factory/fullSyncRequestersContainerFactory.go]: Right, removed

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**